cms/static/js/models/xblock_info.js

```diff
@@ -172,13 +182,33 @@ function(Backbone, _, str, ModuleUtils) {
             return !this.get('published') || this.get('has_changes');
         },
 
-        canBeDeleted: function(){
-            //get the type of xblock
-            if(this.get('override_type') != null) {
-                var type = this.get('override_type');
+        isDeletable: function() {
+            return this.isActionRequired('deletable');
+        },
+
+        isDraggable: function() {
+            return this.isActionRequired('draggable');
+        },
 
-                //hide/remove the delete trash icon if type is entrance exam.
-                if (_.has(type, 'is_entrance_exam') && type['is_entrance_exam']) {
+        isChildAddable: function(){
+            return this.isActionRequired('childAddable');
+        },
+
+        isHeaderVisible: function(){
+            if(this.get('is_header_visible')!= null) {
```

Comment:
For JavaScript, use !== and === instead of != and == (so that it does not do type coercion). If you are not using an editor that does JS linting, I would recommend that you change to one that does, or investigate how to run jslint in edx-platform (it was added by some developers who are no longer here, and I'm not sure how to run it).